### ISSUE TITLE
feat: improve expo example

### ISSUE THREAD

**Issue by juliusmarminge** [t3-oss/create-t3-turbo#851]
### Describe the feature you'd like to request

the idea of the expo example was never to have a perfect mobile setup from the start, but more to show the splitting possible.

but i tihnk we can add some stuff that will enhance the user experience of mobile devs checking this out

### Describe the solution you'd like to see

- light/dark mode?
- modals/drawers?
- auth
- more??

**Comment by EuanMorgan** [comment]
Could possibly add Apple sign in to this?
Just because its in their ToS now that apps must offer it if they use any third party auth like the discord provider here

**Comment by juliusmarminge** [comment]
Yup - i just dont have an Apple developer account so cant test it 😆

**Comment by trevorpfiz** [comment]
I am working on this. thoughts on using [react-native-reusables](https://github.com/mrzachnugent/react-native-reusables)? best 'shadcn/ui for React Native' I have found.

**Comment by trevorpfiz** [comment]
> Could possibly add Apple sign in to this? Just because its in Apple's ToS now that apps must offer it if they use any third party auth like the discord provider here

I am trying to add Apple sign-in, but running into two issues:

1. https://github.com/nextauthjs/next-auth/issues/9989
2. https://github.com/nextauthjs/next-auth/discussions/6898#discussioncomment-8429480
